services/smtp-bridge-emulator/apihandlers/send-email.go

```diff
@@ -0,0 +1,187 @@
+func createEmailsDirectory() error {
+	if err := os.MkdirAll(EMAILS_DIR, os.ModePerm); err != nil {
+		slog.Error("Error creating directory: "+EMAILS_DIR, slog.String("error", err.Error()))
+		return err
+	}
+	return nil
+}
+
+// saves an email to a .eml file
+func saveEmailAsEML(email SendEmailReq) error {
+	if err := createEmailsDirectory(); err != nil {
+		return err
+	}
+
+	// Generate a unique EML file path
+	emlFilePath, err := getUniqueEMLFilePath()
+	if err != nil {
+		slog.Error("Error generating EML file path", slog.String("FilePath", emlFilePath), slog.String("error", err.Error()))
+		return err
+	}
+
+	// Create .eml file
+	file, err := os.Create(emlFilePath)
+	if err != nil {
+		slog.Error("Failed to create EML file ", slog.String("path", emlFilePath), slog.String("error", err.Error()))
+		return err
+	}
+	defer file.Close()
+
+	//compose email headers and write to file
+	headers := composeHeaders(email)
+	if err := writeHeadersToFile(file, headers); err != nil {
+		slog.Error("Failed to write headers to file", slog.String("FilePath", emlFilePath), slog.String("error", err.Error()))
+		return err
+	}
+
+	// Write content (body) to file
+	_, err = file.WriteString(email.Content)
+	if err != nil {
+		slog.Error("Failed to write email content", slog.String("error", err.Error()))
+		return err
+	}
+
+	slog.Info("Email has been saved as EML file successfully")
+	return nil
+}
+
+// constructs the email headers as a map from SendEmailReq
+func composeHeaders(email SendEmailReq) map[string]string {
+	headers := map[string]string{
+		"To":      strings.Join(email.To, ", "),
+		"Subject": email.Subject,
+		"Date":    time.Now().Format(time.RFC1123Z),
+	}
+
+	// Priority header
+	if email.HighPrio {
+		headers["Priority"] = "High"
+	}
+
+	// Optional header overrides
+	if email.HeaderOverrides != nil {
+		if email.HeaderOverrides.From != "" {
+			headers["From"] = email.HeaderOverrides.From
+		}
+		if email.HeaderOverrides.Sender != "" {
+			headers["Sender"] = email.HeaderOverrides.Sender
+		}
+		if !email.HeaderOverrides.NoReplyTo && len(email.HeaderOverrides.ReplyTo) > 0 {
+			headers["Reply-To"] = strings.Join(email.HeaderOverrides.ReplyTo, ", ")
+		}
+	}
+
+	return headers
+}
+
+// writes headers map to the provided file
+func writeHeadersToFile(file *os.File, headers map[string]string) error {
+	for key, value := range headers {
+		if _, err := file.WriteString(key + ": " + value + "\n"); err != nil {
+			slog.Error("failed to write header", slog.String("key", key), slog.String("error", err.Error()))
+			return err
+		}
+	}
+	// Separate headers from body
+	if _, err := file.WriteString("\n"); err != nil {
+		slog.Error("Failed to write header-body separator", slog.String("error", err.Error()))
+		return err
+	}
+	return nil
+}
+
+// returns a unique file path for the EML file i.e. timestamp.eml, but appending a counter if needed.
+func getUniqueEMLFilePath() (string, error) {
```

Comment:
I would keep the emails still organised per email address, similarly as you've did that before --> so create on folder per "to" and store the EML file into these folders. It makes much easier to find the right email in the folder, when testing.